app.rb

```diff
@@ -17,7 +17,24 @@ class OpenidConnectRelyingParty < Sinatra::Base
   BASIC_AUTH = { username: ENV['IDP_USER'], password: ENV['IDP_PASSWORD'] }.freeze
 
   get '/' do
-    authorization_url = openid_configuration[:authorization_endpoint] + '?' + {
+    if openid_configuration
+      erb :index, locals: { authorization_url: authorization_url }
+    else
+      erb :errors, locals: { error: openid_configuration_error }
+    end
+  end
+
+  get '/auth/result' do
+    token_response = token(params[:code])
+    userinfo_response = userinfo(token_response[:id_token])
+
+    erb :success, locals: { userinfo: userinfo_response }
```

Comment:
Should we add a rescue and render an error in case something false? Ex if the `token` call fails?